backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/barchart/internal/BarChartTitle.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.barchart.internal
+
+import androidx.compose.animation.AnimatedContent
+import androidx.compose.animation.AnimatedContentScope
+import androidx.compose.animation.ContentTransform
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import net.skyscanner.backpack.compose.barchart.BpkBarChartModel
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.utils.ScrollingDirection
+import net.skyscanner.backpack.compose.utils.lastScrollingDirection
+
+@OptIn(ExperimentalAnimationApi::class)
```

Comment:
what's experimental here? we should generally avoid using experimental apis as they'll likely change in the future. this is unavoidable in some cases where the material component isn't stable yet, but in an animation case relying on experimental apis is risky

Reply:
AnimatedContent() is experimental. It's also the way the content animation should be performed according to this diagram: https://developer.android.com/jetpack/compose/animation

A lot of animation APIs are experimental in Compose. It doesn't mean the code is buggy or not ready to be used – it just means the API might be changed. In most of the cases it'll be deprecated first, so we'll have enough time to migrate away. We already use this animation in some other places, I wouldn't be worried about that.

Besides, in worth case scenario this specific usage can just be replaced with Crossfade or have no animation at all.